src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -1556,7 +1556,7 @@ address StubGenerator::generate_sha256_implCompress(bool multi_block, const char
 }
 
 address StubGenerator::generate_sha512_implCompress(bool multi_block, const char *name) {
-  assert(VM_Version::supports_avx2(), "");
+  assert(VM_Version::supports_avx2() || VM_Version::supports_sha512(), "");
   assert(VM_Version::supports_bmi2(), "");
```

Comment:
The SHA512 algorithm is also using AVX2 instructions but doesn't need bmi2.
So this needs to be changed to:
 assert(VM_Version::supports_avx2(), "");
 assert(VM_Version::supports_bmi2()  || VM_Version::supports_sha512(), "");